src/Hub.php

```diff
@@ -83,31 +87,63 @@ public function getFactory(): ?TokenFactoryInterface
      */
     public function publish(Update $update): string
     {
-        $postData = [
-            'topic' => $update->getTopics(),
-            'data' => $update->getData(),
-            'private' => $update->isPrivate() ? 'on' : null,
-            'id' => $update->getId(),
-            'type' => $update->getType(),
-            'retry' => $update->getRetry(),
-        ];
-
         $jwt = $this->getProvider()->getJwt();
         $this->validateJwt($jwt);
+        $postData = $this->updateToArray($update);
 
         try {
-            return $this->httpClient->request('POST', $this->getUrl(), [
+            $response = $this->httpClient->request('POST', $this->getUrl(), [
                 'auth_bearer' => $jwt,
                 'headers' => [
                     'Content-Type' => 'application/x-www-form-urlencoded',
                 ],
                 'body' => Internal\QueryBuilder::build($postData),
             ])->getContent();
+
+            $this->logUpdate($update);
```

Comment:
I would move the null check here to prevent a useless function call in case the logger isn't injected (this code may be in the hot path if the app sends a lof of updates):


```suggestion
            if (null !== $this->logger) {
                $this->logUpdate($update);
            }
```